### ISSUE TITLE
An error occurs depending on the location of the collection parameter. 

### ISSUE THREAD

**Issue by DaeAkin** [mapstruct/mapstruct#2276]
DemoDto.java
```java
@Getter
@Setter
@ToString
public class DemoDto {
    private String userId;
    private String name;
}
```

DemoEntity.java
```java
@Getter
@Setter
@ToString
public class DemoEntity {
    private String userId;
    private String name;

    List<String> someList;
}
```

DemoMapper.java
```java
@Mapper(componentModel = "spring",
        nullValueMappingStrategy = NullValueMappingStrategy.RETURN_DEFAULT)
public interface DemoMapper {
    DemoEntity toDto(List<String> someList,DemoDto demoDto);
}
```

when compile this class,  An error occurs :   error: Can't generate mapping method from iterable type to non-iterable type.
    DemoEntity toDto(List<String> someList,DemoDto demoDto);

but I change the position `List<String> someList` to behind, it works.

```java
@Mapper(componentModel = "spring",
        nullValueMappingStrategy = NullValueMappingStrategy.RETURN_DEFAULT)
public interface DemoMapper {
    DemoEntity toDto(DemoDto demoDto,List<String> someList);
}
```

I'm using this : 
```
    compileOnly group: 'org.mapstruct', name: 'mapstruct', version: '1.3.1.Final'
    annotationProcessor group: 'org.mapstruct', name: 'mapstruct-processor', version: '1.3.1.Final'

    compileOnly  group: 'org.projectlombok', name: 'lombok', version: '1.18.12'
    annotationProcessor   group: 'org.projectlombok', name: 'lombok', version: '1.18.12'
```

Is it Issue? Thanks.

**Comment by filiphr** [comment]
This is a duplicate of #607 which was recently implemented and will be part of 1.5.0.

In order to make it work in your example you'll have to swap the source paremters.

e.g.

```
@Mapper(componentModel = "spring",
        nullValueMappingStrategy = NullValueMappingStrategy.RETURN_DEFAULT)
public interface DemoMapper {
    DemoEntity toDto(DemoDto demoDto, List<String> someList);
}

```

**Comment by DaeAkin** [comment]
Thanks for answering.
